src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -66,16 +81,41 @@ public synchronized DriverCommand getCurrentCommand() {
 
 	public synchronized void clearCurrentCommand() {
 		currentCommand = null;
+		names.clear();
+	}
+	public boolean getMacroState(){
+		return macroState;
+	}
+	public void changeMacroState(){
+		macroState=!macroState;
 	}
-
 	public synchronized String getCurrentCommandString() {
-		if (getCurrentCommand() == null) {
+		if (names.size()==0) {
 			return "No command loaded";
 		} else
-			return getCurrentCommand().toString();
+			return names.toString();
 	}
 
 	public Publisher getChangePublisher() {
 		return changePublisher;
 	}
+
+	@Override
+	public void setPosition(int i, int i1) {
+		commands.add(new SetPositionCommand(i, i1));
+	}
+
+	@Override
+	public void operateTo(int i, int i1) {
+		commands.add(new OperateToCommand(i, i1));
+	}
+	public List<DriverCommand> getCommands(){
+		return commands;
+
+	}
+	public void clearCommands(){
+		commands.clear();
+		names.clear();
+	}
```

Comment:
Try extracting this part to a separate driver class and use it from the manager.